package/src/contexts/messageInputContext/MessageInputContext.tsx

```diff
@@ -1039,26 +994,30 @@ export const MessageInputProvider = ({
           if (message && isBouncedMessage(message)) {
             await removeMessage(message);
           }
+
           value.sendMessage({
             attachments,
-            mentioned_users: uniq(mentionedUsers),
+            // TODO: Handle unique users
+            mentioned_users: localMessage.mentioned_users?.map((user) => user.id),
             /** Parent message id - in case of thread */
             parent_id: thread?.id,
-            quoted_message_id: value.quotedMessage ? value.quotedMessage.id : undefined,
+            poll_id: localMessage.poll_id,
+            quoted_message_id: localMessage.quoted_message_id,
             show_in_channel: sendThreadMessageInChannel || undefined,
-            text: prevText,
+            text: localMessage.text,
             ...customMessageData,
-          } as unknown as StreamMessage);
+          });
 
-          value.clearQuotedMessageState();
+          // TODO: This might not be needed. Think about it.
+          // value.clearQuotedMessageState();
```

Comment:
Probably not needed, since this is now handled by the composer itself and clearing should take care of that too. But check please

Reply:
This is removed in attachment manager PR as well.